Material.CREAKING_HEAR itemstack exception

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest version of MockBukkit?

- [X] I am using the latest version of MockBukkit.

### Minecraft Version

1.21

### Describe the bug

```
org.mockbukkit.mockbukkit.exception.ItemMetaInitException: Failed to instanciate item meta class class org.mockbukkit.mockbukkit.inventory.meta.BlockStateMetaMock
	at org.mockbukkit.mockbukkit.inventory.ItemStackMock.findItemMeta(ItemStackMock.java:406)
	at org.mockbukkit.mockbukkit.inventory.ItemStackMock.<init>(ItemStackMock.java:64)
	at org.mockbukkit.mockbukkit.inventory.ItemTypeMock.createItemStack(ItemTypeMock.java:157)
	at org.mockbukkit.mockbukkit.inventory.ItemTypeMock.createItemStack(ItemTypeMock.java:151)
	at me.glicz.skanalyzer.SkAnalyzer.main(SkAnalyzer.java:68)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:74)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
	at org.mockbukkit.mockbukkit.inventory.ItemStackMock.findItemMeta(ItemStackMock.java:398)
	... 4 more
Caused by: java.lang.UnsupportedOperationException: 'CREAKING_HEART' is not known to have a BlockStateMeta ItemMeta type
	at org.mockbukkit.mockbukkit.inventory.meta.BlockStateMetaMock.<init>(BlockStateMetaMock.java:135)
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	... 7 more
```

### Reproducible Test

```java
MockBukkit.mock();
Material.CREAKING_HEART.asItemType().createItemStack();
```

### Anything else?

_No response_

--- a/src/main/java/org/mockbukkit/mockbukkit/inventory/meta/BlockStateMetaMock.java
+++ b/src/main/java/org/mockbukkit/mockbukkit/inventory/meta/BlockStateMetaMock.java
@@ -71,6 +71,7 @@ public class BlockStateMetaMock extends ItemMetaMock implements BlockStateMeta
 
 	static
 	{
+		// To update this list check Paper class "CraftItemMetas"
 		Map<Material, Class<? extends TileStateMock>> map = new HashMap<>();
 		MaterialTags.SHULKER_BOXES.getValues().forEach(m -> map.put(m, ShulkerBoxStateMock.class));
 		MaterialTags.SIGNS.getValues().forEach(m -> map.put(m, SignStateMock.class));
@@ -86,6 +87,7 @@ public class BlockStateMetaMock extends ItemMetaMock implements BlockStateMeta
 		map.put(Material.CHEST, ChestStateMock.class);
 		map.put(Material.CHISELED_BOOKSHELF, null);
 		map.put(Material.COMMAND_BLOCK, CommandBlockStateMock.class);
+		map.put(Material.CREAKING_HEART, null);
 		map.put(Material.CHAIN_COMMAND_BLOCK, CommandBlockStateMock.class);
 		map.put(Material.REPEATING_COMMAND_BLOCK, CommandBlockStateMock.class);
 		map.put(Material.COMPARATOR, ComparatorStateMock.class);

--- a/src/test/java/org/mockbukkit/mockbukkit/inventory/meta/BlockStateMetaMockTest.java
+++ b/src/test/java/org/mockbukkit/mockbukkit/inventory/meta/BlockStateMetaMockTest.java
@@ -1,18 +1,18 @@
 package org.mockbukkit.mockbukkit.inventory.meta;
 
 import org.bukkit.Material;
+import org.bukkit.Server;
 import org.bukkit.block.BlockState;
 import org.bukkit.block.Container;
 import org.bukkit.inventory.Inventory;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.inventory.meta.BlockStateMeta;
+import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
 import org.mockbukkit.mockbukkit.MockBukkitExtension;
 import org.mockbukkit.mockbukkit.MockBukkitInject;
-import org.bukkit.Server;
-import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockbukkit.mockbukkit.block.state.AbstractFurnaceStateMock;
 import org.mockbukkit.mockbukkit.block.state.ContainerStateMock;
 import org.mockbukkit.mockbukkit.inventory.ItemStackMock;
@@ -20,7 +20,12 @@ import org.mockbukkit.mockbukkit.inventory.ItemStackMock;
 import java.util.Map;
 import java.util.stream.Stream;
 
-import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNotSame;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 @ExtendWith(MockBukkitExtension.class)
 public class BlockStateMetaMockTest
@@ -148,6 +153,22 @@ public class BlockStateMetaMockTest
 		ItemStack item = ItemStack.of(type);
 		assertInstanceOf(BlockStateMetaMock.class, item.getItemMeta());
 		assertInstanceOf(ContainerStateMock.class, ((BlockStateMetaMock) item.getItemMeta()).getBlockState());
+	}
+
+	@ParameterizedTest
+	@MethodSource("getPossibleItemTypes")
+	void givenPossibleBlockMaterials(Material expected)
+	{
+		ItemStack item = expected.asItemType().createItemStack();
+		assertNotNull(item);
+		assertEquals(expected, item.getType());
+	}
+
+	public static Stream<Arguments> getPossibleItemTypes()
+	{
+		return Stream.of(Material.values())
+				.filter(m -> m.asItemType() != null)
+				.map(Arguments::of);
 	}
 
 	public static Stream<Arguments> container_Materials()

